src/item_pocket.h

```diff
@@ -0,0 +1,277 @@
+        /**
+         * Is part of the recursive call of item::process. see that function for additional comments
+         * NOTE: this destroys the items that get processed
+         */
+        void process( player *carrier, const tripoint &pos, bool activate, float insulation = 1,
+                      temperature_flag flag = temperature_flag::TEMP_NORMAL, float spoil_multiplier = 1.0f );
+        pocket_type saved_type() const {
+            return _saved_type;
+        }
+
+        // tries to put an item in the pocket. returns false if failure
+        ret_val<contain_code> insert_item( const item &it );
+        /**
+          * adds an item to the pocket with no checks
+          * may create a new pocket
+          */
+        void add( const item &it );
+        /** fills the pocket to the brim with the item */
+        void fill_with( item contained );
+        bool can_unload_liquid() const;
+
+        // only available to help with migration from previous usage of std::list<item>
+        std::list<item> &edit_contents();
+
+        void migrate_item( item &obj, const std::set<itype_id> &migrations );
+
+        // cost of getting an item from this pocket
+        // @TODO: make move cost vary based on other contained items
+        int obtain_cost( const item &it ) const;
+
+        // this is used for the visitable interface. returns true if no further visiting is required
+        bool remove_internal( const std::function<bool( item & )> &filter,
+                              int &count, std::list<item> &res );
+        // @relates visitable
+        VisitResponse visit_contents( const std::function<VisitResponse( item *, item * )> &func,
+                                      item *parent = nullptr );
+
+        void general_info( std::vector<iteminfo> &info, int pocket_number, bool disp_pocket_number ) const;
+        void contents_info( std::vector<iteminfo> &info, int pocket_number, bool disp_pocket_number ) const;
+
+        void serialize( JsonOut &json ) const;
+        void deserialize( JsonIn &jsin );
+
+        bool same_contents( const item_pocket &rhs ) const;
+        /** stacks like items inside the pocket */
+        void restack();
+        bool has_item_stacks_with( const item &it ) const;
+
+        bool better_pocket( const item_pocket &rhs, const item &it ) const;
+
+        bool operator==( const item_pocket &rhs ) const;
+    private:
+        // the type of pocket, saved to json
+        pocket_type _saved_type = pocket_type::LAST;
+        const pocket_data *data = nullptr;
+        // the items inside the pocket
+        std::list<item> contents;
+        bool _sealed = true;
+};
+
+// an object that has data on how many items can exist inside an item
+struct item_number_overrides {
+    bool was_loaded;
+    // if false any of the other data is useless.
+    // loading any data changes this to true
+    bool has_override = false;
+
+    int num_items = 0;
```

Comment:
For magazines and possibly others that define `ammo_restriction` this should allow different capacities per ammotype.

Reply:
i feel like that's a magazine-specific feature request maybe? i'll toss around some ideas for how to implement it in such a way that i can have an item number override without an ammo restriction.

Reply:
Yeah the use case I have in mind is for magazines that fit different numbers of different calibers. If it would be hard to add in this first round of changes I can handle it later.

Reply:
it wouldn't be hard so much, just waffling over a decision. i want to make sure i leave you with something you don't have to do shuffle around too much